client/state/analytics/actions/enhance-with-site-main-product.js

```diff
@@ -0,0 +1,51 @@
+import { set } from 'lodash';
+import { hasGSuiteWithUs } from 'calypso/lib/gsuite';
+import { hasTitanMailWithUs } from 'calypso/lib/titan';
+import { ANALYTICS_PAGE_VIEW_RECORD } from 'calypso/state/action-types';
+import isDomainOnly from 'calypso/state/selectors/is-domain-only-site';
+import { getDomainsBySiteId } from 'calypso/state/sites/domains/selectors';
+import { getSelectedSite, getSelectedSiteId } from 'calypso/state/ui/selectors';
+
+/**
+ * Enhances any Redux action that denotes the recording of a page view analytics event with an additional property
+ * to specifies the main product of this site:
+ * `domain` - domain only, without email subscription
+ * `email` - domain only with email subscription
+ * `site` - regular site
+ *
+ * @param {object} action - Redux action as a plain object
+ * @param {Function} getState - Redux function that can be used to retrieve the current state tree
+ * @returns {import('redux').AnyAction} the new Redux action
+ * @see client/state/utils/withEnhancers
+ */
+export function enhanceWithSiteMainProduct( action, getState ) {
+	if ( action.type === ANALYTICS_PAGE_VIEW_RECORD ) {
+		const state = getState();
+		const site = getSelectedSite( state );
+
+		if ( site !== null ) {
+			const siteId = getSelectedSiteId( state, site );
+			let mainProduct = 'site';
+
+			if ( isDomainOnly( state, siteId ) ) {
+				mainProduct = 'domain';
+
+				const nonWPCOMDomains = getDomainsBySiteId( state, siteId ).filter(
```

Comment:
We only seem to need the first non-WP.com domain, so I'd suggest using `Array.prototype.find()` instead of `Array.prototype.filter()` here. That way we could directly use the result, rather than having to further call `[ 0 ]` on it.